packages/components/src/popover/index.js

```diff
@@ -364,7 +364,7 @@ const Popover = (
 			refs.floating.current,
 			update
 		);
-	}, [ anchorRef, anchorRect, getAnchorRect ] );
+	}, [ anchorRef, anchorRect, getAnchorRect, update ] );
```

Comment:
For good measure, I ran `npx eslint --rule 'react-hooks/exhaustive-deps: warn' packages/components/src/popover/`.

While on `trunk` it flagged the missing `update` dependency here (and on the `useLayoutEffect` below), it also flagged `reference` and  `refs.floating` as missing dependencies. Since `reference` is a callback ref, and `refs.floating` is a "modern" React ref, I suggest that we add a comment disabling the ESLint rule and explaining the reason why:

```suggestion
		// 'reference' and 'refs.floating' are refs and don't need to be listed
		// as dependencies (see https://github.com/WordPress/gutenberg/pull/41612)
		// eslint-disable-next-line react-hooks/exhaustive-deps
	}, [ anchorRef, anchorRect, getAnchorRect, update ] );
```